homeassistant/components/homewizard_energy/config_flow.py

```diff
@@ -0,0 +1,187 @@
+
+        initialized = False
+        try:
+            with async_timeout.timeout(10):
+                await energy_api.initialize()
+                if energy_api.device is not None:
+                    initialized = True
+
+        except aiohwenergy.DisabledError:
+            _LOGGER.error("API disabled, API must be enabled in the app")
+            return self.async_abort(reason="api_not_enabled")
+
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.error(
+                "Error connecting with Energy Device at %s",
+                entry_info[CONF_IP_ADDRESS],
+            )
+            return self.async_abort(reason="unknown_error")
+
+        finally:
+            await energy_api.close()
+
+        if not initialized:
+            _LOGGER.error("Initialization failed")
+            return self.async_abort(reason="unknown_error")
+
+        # Validate metadata
+        if energy_api.device.api_version != "v1":
+            return self.async_abort(reason="unsupported_api_version")
+
+        if energy_api.device.product_type not in SUPPORTED_DEVICES:
+            _LOGGER.error(
+                "Device (%s) not supported by integration",
+                energy_api.device.product_type,
+            )
+            return self.async_abort(reason="device_not_supported")
+
+        # Configure device
+        entry_info["product_name"] = energy_api.device.product_name
+        entry_info["product_type"] = energy_api.device.product_type
+        entry_info["serial"] = energy_api.device.serial
+
+        self.context[CONF_HOST] = entry_info[CONF_IP_ADDRESS]
+        self.context[CONF_PORT] = entry_info[CONF_PORT]
+        self.context["product_name"] = entry_info["product_name"]
+        self.context["product_type"] = entry_info["product_type"]
+        self.context["serial"] = entry_info["serial"]
+        self.context[
+            "unique_id"
+        ] = f"{entry_info['product_type']}_{entry_info['serial']}"
+        self.context[
+            "name"
+        ] = f"{self.context['product_name']} ({self.context['serial'][-6:]})"
+
+        await self.async_set_unique_id(self.context["unique_id"])
+        self._abort_if_unique_id_configured(updates=entry_info)
+
+        self.context["title_placeholders"] = {
+            "name": self.context["name"],
+            "unique_id": self.context["unique_id"],
+        }
+
+        return await self.async_step_confirm()
+
+    async def async_step_confirm(self, user_input=None):
+        """Handle user-confirmation of node."""
+
+        _LOGGER.debug("config_flow async_step_confirm")
+
+        if user_input is None:
+            return self.async_show_form(
+                step_id="confirm",
+                description_placeholders={"name": self.context["product_name"]},
+                data_schema=Schema(
+                    {
+                        Required("name", default=self.context["product_name"]): All(
```

Comment:
No need to ask for the name. Entries can be renamed in the UI later.

Reply:
I implemented this to have a useful entity name when you add more that one energy device. Otherwise you have to rename all the entities to something like `sensor.refrigerator_active_power` etc.

Quitte a few users that use the custom integration told me that they like to have the name to be added to help with the problem above. It is there a better way to do that?

It is currently not possible to get the name the user had entered from the app, so we can't use that at this moment.

Reply:
When renaming a device in the UI the entities are renamed too.

Reply:
Cool, didn't know that.

Reply:
Removed custom name stuff with https://github.com/home-assistant/core/pull/55812/commits/79f574f2d05a226bddeb6ec3016f667565132e72